book/src/development/define_lints.md

```diff
@@ -0,0 +1,189 @@
+# Define New Lints
+
+The first step in the journey of a new lint is to define the definition
+and registration of the lint in Clippy's codebase.
+We can use the Clippy dev tools to handle this step since setting up the 
+lint involves some boilerplate code.
+
+## Name the Lint
+
+A wise software engineer Phil Karlton once said:
+> There are only two hard things in Computer Science: cache invalidation and naming things.
+
+Naming a lint is no exception.
+Therefore, in case of uncertainty about if the name you chose fits the lint,
+please do the following:
+
+1. Consult our [lint naming guidelines][lint_naming]
+2. Ping a [Clippy team member][clippy_team_members] in our [Zulip] chat
+3. Comment on the corresponding Github issue (less preferrable as comments might be overlooked)
+
+For now, let's imagine that your fellow developers at work tend to define some of their
+functions with the name `foo` and forget to re-name it to a more meaningful name
+when they submit a pull request.
+
+`foo` is a highly non-descriptive name for a function, so we want to detect this
+bad naming and fix it early on in the development process.
+
+For this, we will create a lint that detects these `foo` functions and
+help our fellow developers correct this bad practice. Note that in Clippy,
+lints are generally written in snake cases.
+We can name this new lint `foo_functions`.
+
+## Add and Register the Lint
```

Comment:
This chapter is a good example on how this part of the documentation should be written. Just an explanation for how the `cargo dev new_lint` is used and what the expected output is.